plugin/tidal.vim

```diff
@@ -43,58 +81,68 @@ endfunction
 
 let s:tidal_term = -1
 
+" NVim and VIM8 Terminal Implementation
+" =====================================
 function! s:TerminalOpen()
-  if !has('nvim')
-    echom "'terminal' target currently supported on NeoVim only. Use 'tmux'"
-    return
-  endif
-
   if s:tidal_term != -1
     return
   endif
 
-  split term://tidal
-
-  let s:tidal_term = b:terminal_job_id
-
-  " Give tidal a moment to start up so the command doesn't show up at the top
-  " unaesthetically.
-  " But this isn't very robust.
-  sleep 500m
-
-  " Make terminal scroll to follow output
-  :exe "normal G"
-
-  " Make small & on the bottom.
-  :exe "normal \<c-w>J"
-  :exe "normal \<c-w>\<c-w>"
-  :exe "normal \<c-w>_"
-  :exe "normal \<c-w>10-"
+  if has('nvim')
+    split term://tidal
+    let s:tidal_term = b:terminal_job_id
+
+    " Give tidal a moment to start up so the command doesn't show up at the top
+    " unaesthetically.
+    " But this isn't very robust.
+    sleep 500m
+
+    " Make terminal scroll to follow output
+    :exe "normal G"
+
+    " Make small & on the bottom.
+    :exe "normal \<c-w>J"
+    :exe "normal \<c-w>\<c-w>"
+    :exe "normal \<c-w>_"
+    :exe "normal \<c-w>10-"
+
+  elseif  has('terminal')
+    let startup = s:parent_path . "/Tidal.ghci"
+    execute "below split"
+    let s:tidal_term = term_start((g:tidal_ghci . " -ghci-script=" . startup), #{
+          \ term_name: 'tidal',
+          \ term_rows: 10,
+          \ norestore: 1,
+          \ curwin: 1,
+          \ })
+    wincmd p " return focus to previous buffer
+  endif
 endfunction
 
 function! s:TerminalSend(config, text)
   call s:TerminalOpen()
-  call jobsend(s:tidal_term, a:text)
+  if has('nvim')
+    call jobsend(s:tidal_term, a:text . "\<CR>")
+  elseif has('terminal')
+    call term_sendkeys(s:tidal_term, a:text . "\<CR>")
+  endif
 endfunction
 
 " These two are unnecessary AFAIK.
 function! s:TerminalPaneNames(A,L,P)
 endfunction
+
 function! s:TerminalConfig() abort
 endfunction
 
+
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 " Helpers
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
-function! s:SID()
-  return matchstr(expand('<sfile>'), '<SNR>\zs\d\+\ze_SID$')
-endfun
-
-function! s:WritePasteFile(text)
```

Comment:
Removing this helper function breaks tmux support because it can't send the code to the split with the REPL.